### ISSUE TITLE
inconsistent documentation, t.i.task.coiterate

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5632]
|<img alt="pagga's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| pagga reported|
|-|-|
|Trac ID|trac#5632|
|Type|defect|
|Created|2012-04-10 20:37:03Z|
|Branch|https://github.com/twisted/twisted/tree/coiterate-doc-5632|

t.i.task.coiterate returns a Deferred as discussed here [http://stackoverflow.com/questions/10082259/twisted-python-what-is-the-difference-between-cooperate-and-coiterate]. Documentation at [http://twistedmatrix.com/documents/current/api/twisted.internet.task.html#cooperate] does not state this. Documentation for t.i.task.Cooperator at [http://twistedmatrix.com/documents/current/api/twisted.internet.task.Cooperator.html#cooperate] does state this. 

Attachments:

* [5632-coiterate-docstring.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3b0/3b07a8b302314d11ea1e7651a7b9b5e8d804275e/d4cdafcba28f2af746b61d556209f9af54d1e1a3.patch) (564 bytes) - added by amacleod on 2012-06-02 01:21:12Z - Updated docstring for coiterate function in twisted.internet.task module.

<details><summary>Searchable metadata</summary>

```
trac-id__5632 5632
type__defect defect
reporter__pagga pagga
priority__normal normal
milestone__ 
branch__branches_coiterate_doc_5632 branches/coiterate-doc-5632
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation_easy documentation easy
time__1334090223000000 1334090223000000
changetime__1339952086000000 1339952086000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|amacleod set owner to amacleod<br>amacleod set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|amacleod removed owner<br>amacleod set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @therve|
|-|-|

Looking good, applying.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [34575]) Branching to 'coiterate-doc-5632'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [34577]) Merge coiterate-doc-5632

Author: amacleod
Reviewer: therve
Fixes: [#5632](https://github.com/twisted/twisted/issues/5632)

Update docstring of coiterate to mention the fact that it returns a Deferred.
